pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -900,10 +902,12 @@ remediateIfNeeded will try to remediate unhealthy machines (annotated by MHC) by
 The full remediation flow is:
 1) Power off the host
 2) Add poweredOffForRemediation annotation to the unhealthy Machine
-3) Delete the node
-4) Power on the host
-5) Wait for the node the come up (by waiting for the node to be registered in the cluster)
-6) Remove poweredOffForRemediation annotation and the MAO's machine unhealthy annotation
+3) Store node's annotations and labels in Machine's annotations
+4) Delete the node
+5) Power on the host
+6) Wait for the node the come up (by waiting for the node to be registered in the cluster)
```

Comment:
Since this controller is not watching Nodes, what causes it to reconcile when a Node comes up?

Reply:
Not sure how this worked so far. I think that I saw the `Update()` function getting called periodically in the logs (maybe once in 10s).
Anyway, I've added a watch for Node